src/components/ContactUsForm/index.tsx

```diff
@@ -0,0 +1,189 @@
+
+  const handleChange = (key: string, value: string) => {
+    setData((d) => ({ ...d, [key]: value }));
+  };
+
+  const hasRequiredFields = () => {
+    return data.name && data.email && data.subject && data.message;
+  };
+
+  const handleSuccessfulRequest = () => {
+    setData(INITIAL_STATE);
+    window.alert('Message sent successfully');
+  };
+
+  const handleSubmit = async (event: React.SyntheticEvent) => {
+    event.preventDefault();
+
+    if (isLoading) return;
+
+    if (!hasRequiredFields())
+      setErrorMessage('Please fill all the required fields');
+
+    if (typeof fetch === 'undefined' || typeof window === 'undefined') return;
+
+    setIsLoading(true);
+
+    const grecaptcha = (window as any).grecaptcha;
+    let recaptchaToken;
+
+    try {
+      recaptchaToken = await grecaptcha.execute(
+        process.env.GATSBY_RECAPTCHA_V3_SITE_KEY,
+        {
+          action: 'contact-us'
+        }
+      );
+    } catch (error) {
+      console.error(error);
+      setErrorMessage('Something went wrong, please try again');
+    }
+
+    try {
+      const request = await fetch('/api/messages', {
+        method: 'POST',
+        mode: 'same-origin',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          'g-recaptcha-response-data': recaptchaToken,
+          email: data.email,
+          message: data.message,
+          name: data.name,
+          subject: data.subject
+        })
+      });
+      const response = await request.json();
+
+      if (response.status !== 200) setErrorMessage(response.message);
+      if (response.status === 200) handleSuccessfulRequest();
+    } catch (error) {
+      setErrorMessage(error.message);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <section className="flex flex-col justify-center px-x-screen-spacing py-y-screen-spacing lg:p-desktop-screen-spacing bg-gradient-to-b from-blue-200 via-white-200 to-blue-200 lg:to-white-300">
+      <div className="mx-auto max-w-8xl">
+        <h1 className="text-5xl font-bold leading-none text-black lg:text-6xl">
+          Contact Us
+        </h1>
+        <h2 className="my-8 font-sans text-xl font-bold leading-none lg:my-12 text-black-100 lg:text-2xl">
+          This is an organizing space where people in debt gather to Organize,
+          Resist & Reimagine.
+        </h2>
+        <form
+          className="flex flex-col w-full mx-auto md:w-3/4"
+          onSubmit={handleSubmit}
+        >
+          <Input
+            label="Your name:"
+            type={InputType.text}
+            id="name"
+            onChange={(value) => handleChange('name', value)}
+            value={data.name}
+            className="w-full"
+            required
+          />
+          <Input
+            label="Your email:"
+            type={InputType.email}
+            id="email"
+            onChange={(value) => handleChange('email', value)}
```

Comment:
You can use `target.name` to get the name of the input.
https://reactjs.org/docs/forms.html